#!/usr/bin/env python
PACKAGE = "complementary_filter"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add("Auto_detect_contrast_thresholds", bool_t, 0,
"Enable adaptive contrast thresholds (ON/OFF) that self-recalibrate based APS frames. " 
"Warning: ignores user-input for contrast thresholds", False) 

gen.add("High_dynamic_range_mode", bool_t, 0,
"Enable adaptive, per-pixel cutoff_frequency that is sensitive to APS saturation (under/overexposure)", False) 

gen.add("Cutoff_frequency", double_t, 0,
"The cutoff (crossover) frequency of the complementary filter in Hz", 0.5, 0, 20)

gen.add("Contrast_threshold_OFF", double_t, 0,
"Change in log intensity required to trigger OFF event", -0.1, -0.2, 0)

gen.add("Contrast_threshold_ON", double_t, 0,
"Change in log intensity required to trigger ON event", 0.1, 0, 0.2)

gen.add("Intensity_min", double_t, 0, "Minimum displayed intensity", -0.1, -1, 0.5)

gen.add("Intensity_max", double_t, 0, "Maximum displayed intensity", 1.2, 0.5, 2)

gen.add("Spatial_filter_sigma", double_t, 0,
"Sigma parameter for the spatial filter. If sigma = 0, no filtering is applied.", 0, 0, 10)

filter_enum = gen.enum([ gen.const("None", int_t, 0, ""),
                         gen.const("Gaussian", int_t, 1, ""),
                         gen.const("Bilateral", int_t, 2, ""),
                         gen.const("Guided", int_t, 3, "")], "")
gen.add("Spatial_filter", int_t, 0,
"Filter type applied to display images (default None)", 0, 0, 3, edit_method=filter_enum) 

gen.add("Color_display", bool_t, 0,
"Toggle color display.", False)

gen.add("Guide_fade", double_t, 0,
"Fade rate for guided filter guide.", 5, 0, 20)

gen.add("Switch_guide", bool_t, 0,
"Switch guide and image.", False)

gen.add("Guide_eps", double_t, 0,
"'r' factor for guide.", 0.1, 0, 0.4)

gen.add("Guide_s", int_t, 0,
"Subsampling factor for guide. Lower slower", 1, 1, 4)

exit(gen.generate(PACKAGE, "reconstruction_node", "complementary_filter"))
